interfaces/effect_player.hpp

```diff
@@ -82,6 +83,13 @@ namespace bnb::oep::interfaces
          */
         virtual bool call_js_method(const std::string& method, const std::string& param) = 0;
 
+        /**
+         * Evaluate the `script` in effect. This method is thread safe.
+         * @param script JS string to execute
+         * @param result_callback Callback for result, will be called in render thread.
+         */
+        virtual bool eval_js(const std::string& script, const oep_eval_js_result_cb_sptr result_callback) = 0;
```

Comment:
better to use:
virtual bool eval_js(const std::string& script, const oep_eval_js_result_cb& result_callback) = 0;